tests/test_cliUtils.py

```diff
@@ -366,5 +368,60 @@ def test_exist(self):
                 self.assertIn('"foo.txt" should not exist.', result.output)
 
 
+class MWCommandTest(unittest.TestCase):
+    def setUp(self):
+        self.runner = click.testing.CliRunner()
+        self.ctx = None
+
+    def testCaptureOptions(self):
+        """Test that command options are captured in order."""
+
+        @click.command(cls=MWCommand)
+        @click.argument("ARG_A", required=False)
+        @click.argument("ARG_B", required=False)
+        @click.option("-o", "--an-option")
+        @click.option("-s", "--second-option")
+        @click.option("-f", is_flag=True)
+        @click.option("-d/-n", "--do/--do-not", "do_or_do_not")
+        @click.option("--multi", multiple=True)
+        @click.pass_context
+        def cmd(ctx, arg_a, arg_b, an_option, second_option, f, do_or_do_not, multi):
+            self.assertIsNotNone(ctx)
+            self.ctx = ctx
+
+        for args, expected in (
+            ("--an-option foo --second-option bar", None),
+            ("--second-option bar --an-option foo", None),
+            ("--an-option foo", None),
+            ("--second-option bar", None),
+            ("--an-option foo -f --second-option bar", None),
+            ("-o foo -s bar", "--an-option foo --second-option bar"),
+            ("--an-option foo -f -s bar", "--an-option foo -f --second-option bar"),
+            ("--an-option=foo", "--an-option foo"),
+            # NB when using a short flag everything that follows, including an
+            # equals sign, is part of the value!
+            ("-o=foo", "--an-option =foo"),
```

Comment:
I thought the point of this ticket was that the `=` would be handled so why isn't this `--an-option foo` or else a warning issued to indicate that the `=` was likely not intended?